### PR TITLE
PBM-808, PBM-807: CLI - waiting for the logical restore operations

### PR DESCRIPTION
**Pull request by dAdAbird**
- Added "wait" to the PITR restore and Oplog relay.
- Check for the operation heartbeats during the wait. That will prevent indefinite wait if agents are gone. All nodes in non-sharded env - is the only agent that runs a restore.

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Logical | Physical |
| ------------- | ------------- | ------------- |
| New cluster psmdb 4.4 | passed | - |
| New cluster psmdb 4.2 | passed | - |
| Sharded psmdb 4.2 | failed | passed |
| Sharded psmdb 4.4 | failed | passed |
| Non-sharded psmdb 4.2 | passed | failed |
| Single-node psmdb 4.2 | passed | passed |
| Single-node psmdb 4.4 | passed | failed |
| Non-sharded psmdb 4.4 | passed | passed |
| Sharded psmdb 5.0 | failed | failed |
| New cluster psmdb 5.0 | passed | - |
| Non-sharded psmdb 5.0 | failed | failed |
| Single-node psmdb 5.0 | failed | failed |

File: cli/oplog.go [hunk 1]
```diff
@@ -9,6 +10,28 @@ import (
...
+}
+
+func (r oplogReplayResult) String() string {
+	if r.done {
+		return "\nOplog replay successfully finished!\n"
+	}
+	if r.err != "" {
+		return "\n Error: " + r.err
+	}
+	return fmt.Sprintf("Oplog replay %q has started", r.Name)
```
Comment (defbin) [inline cli/oplog.go:32]: `fmt.Stringer` is used to get a string representation of a value that can be used anywhere.
here I see you are using it to format output specifically (~de~coupled) to CLI. I'd use a separate interface/method for CLI output. fallback to `String()` if there is no implementation for CLI output

Comment (defbin) [inline cli/oplog.go:32]: `fmt.Stringer` is used to get a string representation of a value that can be used anywhere.
here I see you are using it to format output specifically (~de~coupled) to CLI. I'd use a separate interface/method for CLI output. fallback to `String()` if there is no implementation for CLI output

**Review by defbin — APPROVED** [review]

File: cli/oplog.go (hunk 1 above)
Comment (dAdAbird) [reply cli/oplog.go:32]: Yes, maybe, need to think about it. But it is used in CLI where the object has to be in plain text, hence string. Maybe we have to have a separate interface which includes Stringer, json.Marshaler etc

**Pull request merged**